src/components/CollectionsProvider.js

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { PRESETS } from "../data";
+import styles from "../App.module.css";
+import useLocalStorageState from '../hooks/useLocalStorageState'
+import sortCollections from '../utils/sortCollections';
+import ProductGrid from './ProductGrid';
+
+export const CollectionsContext = React.createContext();
+
+function CollectionsProvider() {
+
+  // get localStorage collections data if set, grab PRESETS otherwise
+  const [collections, setCollections] = useLocalStorageState('collections', sortCollections(PRESETS));
+
+  // the active collection ID ('beer','wine','soup')
+  const [currentCollectionId, setCurrentCollectionId] = React.useState('');
+
+  // build dropdown options from available collections
+  const collectionOptions = Object.keys(collections).map((key) => {
+    return (
+      <option key={key} value={key}>
+        {collections[key].icon} Calculate cost per {collections[key].baseUnit} of {key}
+      </option>
+    );
+  });
+
+  // handle collection selector
+  function handleCollectionChange(event) {
+    setCurrentCollectionId(event.target.value);
+  }
+
+  // capture current collection prices and update collections in state
+  function updateItemPrice(collectionId, itemId, price) {
+    const nextItems = collections[collectionId].items.map((item) => {
+      if (!price || (item.id === itemId && item.price === price)) {
+        return item;
+      }
+      return item.id === itemId ? { ...item, price: parseFloat(price) } : item;
+    });
+    const nextCollection = { ...collections[collectionId], items: nextItems }
+    const nextCollections = { ...collections, [collectionId]: nextCollection }
+    setCollections(sortCollections(nextCollections));
+  }
+
+  function updateCollection(collectionId, item) {
+    const nextItems = collections[collectionId].items;
+    nextItems.push(item);
+    console.log(nextItems)
+    const nextCollection = { ...collections[collectionId], items: nextItems }
+    const nextCollections = { ...collections, [collectionId]: nextCollection }
+    setCollections(sortCollections(nextCollections));
+  }
+
+  return (
+    <CollectionsContext.Provider value={{ collections, setCollections, currentCollectionId, setCurrentCollectionId, updateItemPrice, updateCollection }}>
```

Comment:
I haven't dug too deeply into this option yet but arguably, one could keep lines 56 to 63 in `App.js` and modify `CollectionsProvider.js` such that

```js
function CollectionsProvider({children}) {

// ...

  const value = useMemo(() => (
    {
      collections,
      setCollections,
      currentCollectionId,
      setCurrentCollectionId,
      updateItemPrice,
      updateCollection
    }
  ), [collections, currentCollectionId, setCollections, updateCollection, updateItemPrice]);

  return (
    <CollectionsContext.Provider value={value}>
      {children}
    </CollectionsContext.Provider>
  )
}
```

So we could keep the responsibility of this file to just the context and not so much the HTML rendering and styling. Though this would involve refactoring and using [useMemo()](https://react.dev/reference/react/useMemo) as above to pass in the correct values to `value`. Also, possibly moving `collectionOptions` and `handleCollectionChange()`.

Reply:
Thanks @arnest00 -- I originally had that markup block in App.js but sort of liked combining the whole collections bit into one place... however, I see your point about minimizing the responsibility of the provider to just context and logic. I'll work on refactoring that piece.

Reply:
@arnest00 take a look at [eed9498](https://github.com/bengoshow/unit-converter/pull/6/commits/eed94980774a8083250d9a4cdd9009d5bc5e03e8) and let me know your thoughts. Is this moving in the direction you were thinking?